test_simplified_configs.py

```diff
@@ -0,0 +1,162 @@
+from typing import List, Optional
+
+from pydantic import BaseModel, ValidationError
+
+from ert3.config import PluginConfigManager
+from ert3.config import ErtBaseModel, ErtPluginField
+
+
+class _StagesConfig(BaseModel):
+    class Config:
+        validate_all = True
+        validate_assignment = True
+        extra = "forbid"
+        allow_mutation = False
+        arbitrary_types_allowed = True
+
+
+class TransformationConfigBase(_StagesConfig):
+    """Common config for all Transformations"""
+
+    location: str
+
+
+class FileTransformationConfig(TransformationConfigBase):
+    mime: str = ""
+
+
+class SummaryTransformationConfig(TransformationConfigBase):
+    smry_keys: Optional[List[str]] = None
+
+
+class DirectoryTransformationConfig(TransformationConfigBase):
+    pass
+
+
+class DummyInstance:
+    """
+    This is dummy instance that just stores the config.
+    This would be the actual Transformations in the real case.
+    """
+
+    def __init__(self, config) -> None:
+        self._config = config
+
+    def __str__(self) -> str:
+        return f"DummyInstance(config={self._config})"
+
+
+def get_configs(pm):
+    """
+    We now need the create configs that use the ErtPluginField dynamically,
+    as we would like to control the schema created based on the plugins we provide.
+    This will allow us to specify a subset of plugins we want to have effect at runtime,
+    such as only using configs from ert in tests.
+    """
+
+    class StageIO(ErtBaseModel, _StagesConfig, plugin_manager=pm):
+        name: str
+        transformation: ErtPluginField
```

Comment:
While not currently implemented I would like the category that you "bind to" to be specified as a parameter to the  ErtPluginField. This way there is no implicit connections between the name of the field and the plugin category, which should make it more flexible as well. This should be fairly easy to implement, it was just not the main focus of the first iteration of the POC :slightly_smiling_face: 
```suggestion
        whatever_you_want: ErtPluginField("transformation")
```